OpenSim/Common/Component.h

```diff
@@ -570,53 +609,83 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
         return getProperty_connectors().size();
     }
 
-    /** Access the number of Inputs that this component has. */
-    int getNumInputs() const {
-        return int(_inputsTable.size());
-    }
-
-    /** Access the number of Outputs that this component has. */
-    int getNumOutputs() const {
-        return int(_outputsTable.size());
-    }
-
-    /** Collect and return the names of Outputs as an std::vector. */
-    std::vector<std::string> getOutputNames() const {
-        std::vector<std::string> names;
-        for (auto it = getOutputsBegin(); it != getOutputsEnd(); it++)
-            names.push_back(it->first);
+    /** Collect and return the names of the connectors in this
+     * component. The names are in the same order as the connectors. */
+    std::vector<std::string> getConnectorNames() {
+        std::vector<std::string> names(getNumConnectors());
+        for (int i = 0; i < getNumConnectors(); ++i) {
+            names[i] = get_connectors(i).getName(); 
+        }
         return names;
     }
 
-    /** Access a read-only Connector by index.
-    @see getNumConnectors()
-     */
-    const AbstractConnector& getConnector(int i) const {
-        return get_connectors(i);
-    }
-
-    /** Access a writeable Connector by index.
-    @see getNumConnectors()
+    /**
+    * Get the "connectee" object that the Component's Connector
+    * is bound to. Guaranteed to be valid only after the Component
+    * has been connected (that is connect() has been invoked).
+    * If the Connector has not been connected, an exception is thrown.
+    *
+    * This method is for getting the concrete connectee object, and is not
+    * available in scripting. If you want generic access to the connectee as an
+    * Object, use the non-templated version.
```

Comment:
No; this method can be used in scripting if it has a default template argument, but I would need to add a line to the SWIG interface file:

```
%template(getConnectee) Component::getConnectee<Component>;
```

I thought a long time about whether I should do that, or do what I chose to do. In the end, I chose to create a separate non-template method:

1. In at least one case, there is good reason for the `C = Component` variant to be implemented differently from the more generic variant.
2. For consistency with (1), then it makes sense for all similar methods to have a non-template variant.
3. Scripting users may be scared by the template function in doxygen, and not think that the function is available in scripting. By creating the non-template function, we are explicit about its existence in scripting.